src/main/java/org/dependencytrack/tasks/repositories/ComposerMetaAnalyzer.java

```diff
@@ -73,76 +103,286 @@ public RepositoryType supportedRepositoryType() {
+
+        packageVersions.names().forEach(item -> {
+            JSONObject packageVersion = packageVersions.getJSONObject((String)item);
+            String version =  packageVersion.getString("version");
+            if (version.startsWith("dev-") || version.endsWith("-dev")) {
+                // dev versions are excluded, since they are not pinned but a VCS-branch.
+                // this case doesn't seem to happen anymore with V2, as dev (untagged) releases are not part of the response anymore
+                return;
+            }
+
+            // Some (old?) repositories like composer.amasty.com/enterprise do not include a 'version_normalized' field
+            // TODO Should we attempt to normalize ourselves? The PHP code uses something that results in 4 parts instead of 3, i.e. 2.3.8.0 instead of 2.3.8. Not sure if that works with Semver4j
+            String version_normalized = packageVersion.getString("version");
+            if (packageVersion.has("version_normalized")) {
+                version_normalized = packageVersion.getString("version_normalized");
+            }
+
+            ComparableVersion currentComparableVersion = new ComparableVersion(version_normalized);
+            if (currentComparableVersion.compareTo(latestVersion) < 0) {
+                // smaller version can be skipped
+                return;
+            }
 
-                final String published = composerPackage.getJSONObject(key).getString("time");
+            latestVersion.parseVersion(stripLeadingV(version_normalized));
+            meta.setLatestVersion(version);
+
+            if (packageVersion.has("time")) {
+                final String published = packageVersion.getString("time");
                 try {
                     meta.setPublishedTimestamp(dateFormat.parse(published));
-                } catch (ParseException e) {
-                    LOGGER.warn("An error occurred while parsing upload time", e);
+                } catch (Exception e) {
+                    LOGGER.warn("An error occurred while parsing time", e);
                 }
-            });
-        } catch (IOException ex) {
-            handleRequestException(LOGGER, ex);
-        } catch (Exception ex) {
-            throw new MetaAnalyzerException(ex);
-        }
-
+            } else {
+                //TODO some repositories like packages.drupal.org include a 'dateStamp' field, example 1700068743
+                // Some repositories like packages.drupal.org and composer.amasty.com/entprise do not include the name field for a version, so print purl
+                LOGGER.warn("Field 'time' not present in metadata for " + component.getPurl());
+            }
+        });
         return meta;
     }
 
+    /**
+     * Clears the cache of repository root metadata.
+     * Needed in unit tests as we currently only have url as a cache key and all unit tests run against localhost...
```

Comment:
With WireMock you can launch the mock server on a dynamic port: https://github.com/DependencyTrack/dependency-track/blob/9a470cda5ef9d5df69684a03f5667b8554e893d0/src/test/java/org/dependencytrack/tasks/scanners/OssIndexAnalysisTaskTest.java#L41-L42

Reply:
This would still require multiple `WireMockRule` instances to have different ports for each Test. So might be easier to use multiple `ClientAndServer` instances. But if we add more info like repository ID to the analyzer we can just extend the cache key with that.

Reply:
Added repositoryID to the cache key.